index.js

```diff
@@ -9,20 +9,23 @@ const client = new Discord.Client();
 let lastKristen = 0;
 
 client.on("ready", () => {
-	console.log(`Logged in as ${client.user.tag}!`);
+	console.log(util.format("Logged in as %s (id %d)", client.user.username, client.user.id));
+	setRandomActivity();
 });
 
 client.on("error", function(err) {
 	console.log(err);
 });
 
 client.on("message", (msg) => {
-	console.log(util.format("User %s (id: %d) wrote: %s", msg.author.username, msg.author.id, msg.content));
-
-	if (msg.author.username === client.user.username) {
+	// Ignore messages from ourselves and other bots to prevent spam
+	if (msg.author.username === client.user.username || msg.author.bot) {
 		return;
 	}
 
+	console.log(util.format("User %s (id: %d) wrote: %s", msg.author.username, msg.author.id, msg.content));
+	setRandomActivity();
```

Comment:
It might be a good idea to add a command for the bot to set its activities instead of having that be tied to whenever anyone on the server speaks.

Reply:
Or, as an alternative, tie the activity back to whenever the bot actually sends a reply instead of anytime the bot parses a message not from a bot.

Reply:
I went ahead and changed the logic so that it only calls setRandomActivty when Haku responds to a command.